electrum/gui/qt/rbf_dialog.py

```diff
@@ -48,19 +48,23 @@ def __init__(
         ok_button = OkButton(self)
         self.adv_button = QPushButton(_("Show advanced settings"))
         warning_label = WWLabel('\n')
-        warning_label.setStyleSheet(ColorScheme.RED.as_stylesheet())
         self.feerate_e = FeerateEdit(lambda: 0)
         self.feerate_e.setAmount(max(old_fee_rate * 1.5, old_fee_rate + 1))
 
         def on_feerate():
             fee_rate = self.feerate_e.get_amount()
             warning_text = '\n'
+            warning_label.setStyleSheet(ColorScheme.DEFAULT.as_stylesheet())
             if fee_rate is not None:
                 try:
                     new_tx = self.rbf_func(fee_rate)
+                    new_inputs = len(new_tx.inputs()) - len(tx.inputs())
+                    if new_inputs > 0:
+                        warning_text = ("({0} new inputs will be added)".format(new_inputs))
```

Comment:
I think it would be better not to guard these three lines by the try-except.
I suggest creating an else branch (try-except-else), and moving these there;
that way, f they raise for whatever reason, the exception can propagate out.

Reply:
UhM, testing this change before committing I noticed this behavior, base tx is 1sat/byte:

- raising to 1 says -> "Cannot bump fee: The new fee rate ...."
- raising to 2 says -> (1 new inputs will be added)
- raising to 3 says -> (2 new inputs will be added)
- raising to 4 it says nothing...

I investigated and yes, starting at 4sat/byte it doesn't add any input but reduces the output. Don't you think that would be nice to add the appropriate message? (output amount will be reduced)

Reply:
Sure, we could do that, feel free to have a look.
Be warned though that that might require a lot more work to do without ugly hacks as it is mostly abstracted away from the GUI code.

Reply:
I tried to do it in the wallet.py code, but since it needs to add another field to transaction class and also is not clean to implement because the transaction is created at the end of the bump functions... I finally created a comparison function which determines if the change, the payment or both have been decreased.